framework/src/util/Utils.cpp

```diff
@@ -154,6 +154,13 @@ std::string GetCurrentWorkingDirectory()
   return std::string();
 }
 
+static const std::string s_CurrWorkingDir = InitCurrentWorkingDirectory();
```

Comment:
It is possible that I have some hidden assumptions in my head 😄 
My assumptions are:

- ```InitCurrentWorkingDirectory``` is only ever called once for the lifetime of the process
- C++11 magic statics can be used
- The function local static is constructed in  ```GetCurrentWorkingDirectory()``` 
- ```InitCurrentWorkingDirectory()``` is only ever called by ```GetCurrentWorkingDirectory()``` 

For a function local static in C++11 (i.e. magic statics), the following would guarantee that ```GetCurrentDirectoryA``` was called only once and only on one thread, even if GetCurrentWorkingDirectory() is called multiple times from multiple threads:

```
std::string& GetCurrentWorkingDirectory()
{
  static std::string s_CurrWorkingDir(InitCurrentWorkingDir());
  return s_CurrWorkingDir;
}
```
On construction of ```s_CurrWorkingDir```, ```InitCurrentWorkingDir()``` will only be called once, no matter how many times ```GetCurrentWorkingDirectory()``` is called since function local statics are only constructed once in a thread-safe way.


As for ```std::call_once``` - after writing some example code for ```std::call_once```, it seems like an inferior choice to function local statics since the current working directory```std::string``` needs to be an in/out parameter to the ```Callable``` object. From what I see, the only significant difference between this and function local statics is that ```std""call_once``` handles exceptions.